lib/pacemaker/pcmk_sched_colocation.c

```diff
@@ -1143,26 +1143,35 @@ pcmk__apply_coloc_to_weights(pe_resource_t *dependent,
     g_hash_table_iter_init(&iter, work);
     while (g_hash_table_iter_next(&iter, NULL, (void **)&node)) {
         if (primary->allocated_to == NULL) {
-            pe_rsc_trace(dependent, "%s: %s@%s -= %d (%s inactive)",
-                         colocation->id, dependent->id, pe__node_name(node),
-                         colocation->score, primary->id);
             node->weight = pcmk__add_scores(-colocation->score, node->weight);
+            pe_rsc_trace(dependent,
+                         "Applied %s to %s score on %s (now %s after "
+                         "subtracting %s because primary %s inactive)",
+                         colocation->id, dependent->id, pe__node_name(node),
+                         pcmk_readable_score(node->weight),
+                         pcmk_readable_score(colocation->score), primary->id);
 
         } else if (pcmk__str_eq(pe_node_attribute_raw(node, attribute), value,
                                 pcmk__str_casei)) {
             if (colocation->score < CRM_SCORE_INFINITY) {
-                pe_rsc_trace(dependent, "%s: %s@%s += %d",
-                             colocation->id, dependent->id,
-                             pe__node_name(node), colocation->score);
                 node->weight = pcmk__add_scores(colocation->score,
                                                 node->weight);
+                pe_rsc_trace(dependent,
+                             "Applied %s to %s score on %s (now %s after "
+                             "adding %s",
+                             colocation->id, dependent->id, pe__node_name(node),
+                             pcmk_readable_score(node->weight),
+                             pcmk_readable_score(colocation->score));
             }
 
         } else if (colocation->score >= CRM_SCORE_INFINITY) {
-            pe_rsc_trace(dependent, "%s: %s@%s -= %d (%s mismatch)",
-                         colocation->id, dependent->id, pe__node_name(node),
-                         colocation->score, attribute);
             node->weight = pcmk__add_scores(-colocation->score, node->weight);
```

Comment:
Likewise interesting that we subtract the score for a mandatory constraint when the attribute doesn't match, and otherwise don't apply the score.

I'm surprised the colocation code hasn't driven you insane yet.

Reply:
That highlights that this can be simplified to `= -CRM_SCORE_INFINITY`, and the log message can simplify as well